tests/integration_test.go

```diff
@@ -0,0 +1,356 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+	"github.com/mikewurtz/taskman/certs"
+	pb "github.com/mikewurtz/taskman/gen/proto"
+	"github.com/mikewurtz/taskman/internal/grpc/server"
+)
+
+const (
+	testUserID     = "client001"
+	testServerAddr = "localhost:50055"
+	testTimeout    = 5 * time.Second
+)
+
+var stopServer func()
+
+func TestMain(m *testing.M) {
+	stopServer = startTestServer()
+
+	code := m.Run()
+
+	if stopServer != nil {
+		stopServer()
+	}
+	os.Exit(code)
+}
+
+func startTestServer() func() {
+	srv, err := server.NewServer(testServerAddr)
+	if err != nil {
+		panic(fmt.Sprintf("failed to start test server: %v", err))
+	}
+
+	go func() {
+		if err := srv.Start(); err != nil {
+			fmt.Printf("Test server error: %v\n", err)
+		}
+	}()
+
+	// Add a small delay to give the server some time to be ready
+	// In a real integration test, we should be using a more reliable/robust mechanism
+	// like a health check endpoint to wait until the server is ready
+	time.Sleep(100 * time.Millisecond)
```

Comment:
It might be even more robust to use something like [`require.Eventually()`](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually) to both save time in the best case (it should usually take well under 100ms to spawn the server) and avoid failing in the worst case (e.g. a very slow CI runner)

Reply:
Sure, we should be able to incorporate that in